### ISSUE TITLE
Ability to pass extra vars to the test playbook

### ISSUE THREAD

**Issue by buster** [goodplay/goodplay#154]
I have a playbook which uses several roles, which in turn use roles as meta/dependencies which in turn define role variables in role/vars/main.yml. 
To execute the tests i need to overwrite a certain variable of a certain role but i don't see a way to do that without touching the production code...
According to http://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html the preference of the role vars is quite high and even higher then vars in the host variables. So setting the variable
in the inventory of the tests doesn't overwrite the variable.

The test playbook contains the `include` in the first line, but i can't see how to pass variables there, nor was i able to use set_fact or include_vars.
For whatever reason, everytime i change the first include line or prepend another task the playbook doesn't execute at all...

Is there any way to pass extra variables to the test playbook so that role vars can be overwritten?
It seems that the only candidates to do so are the following variable types. I suppose some way to apss vars to the "production" playbook would be nice..

- block vars (only for tasks in block)
- task vars (only for the task)
- include_vars
- set_facts / registered vars
- role (and include_role) params
- include params
- extra vars (always win precedence)

**Comment by benjixx** [comment]
> For whatever reason, every time i change the first include line or prepend another task the playbook doesn't execute at all...

This could be a problem with the current ``is_playbook_content`` function. I noticed a similar issue yesterday which I fixed in commit 1bef36f0 but probably the test in this function is not generic enough.

Would you mind to open a separate issue for this problem with an example of when it fails in your case?

**Comment by benjixx** [comment]
Regarding variables, you should be able to use an inventory directory and provide variables within host_vars and group_vars subfolders. Also, I guess using vars in the same play as your roles should do the trick.

I would pretty much like to avoid adding any additional variable handling to goodplay itself as I think Ansible already provides a lot of functionality around that out of the box.

**Comment by buster** [comment]
According to using host_vars.. If i read the precedence correctly, those do not overwrite role vars (see link above).. Inventory variables have a surprisingly low precedence in this list, that's why i'm wondering how to pass variables into the test..
For example, i can imagine doing something like (not sure which of the import/include tasks would fit best here):

```
---
- import_playbook: ../main.yml
  vars:
    dict1:
      key1: val1
```

**Comment by benjixx** [comment]
In case you want to be able to override role variables usually best practice is to set defaults for these variables in ``role/defaults/main.yml``. I tend to use the ``role/vars/main.yml`` file only for variables that are computed variables (usually based on the default variables), e.g. a download url that depends on a version number.

**Comment by benjixx** [comment]
@buster Have you found a solution to your problem? If so, could you please briefly describe it and close this issue?

**Comment by buster** [comment]
@benjixx so the problem was, that i had dependent roles (in meta/main.yml of my roles) which defined variables in (vars/). I wanted to overwrite them, because i'm not able to change the roles.
The workaround was to "import_role" the 3rd party roles from the role (tasks/main.yml) and overwite variables there. This didn't work in Ansible 2.4 due to a bug which was fixed in 2.5,
